src/core/diffusion_grid.h

```diff
@@ -94,20 +96,33 @@ class DiffusionGrid {
     c2_.resize(total_num_boxes_);
     gradients_.resize(3 * total_num_boxes_);
 
+    // If we are utilising the Runge-Kutta method we need to resize an
+    // additional vector, this will be used in estimating the concentration
+    // between diffsuion steps.
+    auto* param = Simulation::GetActive()->GetParam();
+    if (param->diffusion_type_ == "RK") {
+      r1_.resize(total_num_boxes_);
+    }
+
     initialized_ = true;
   }
 
   void ParametersCheck() {
-    // The 1.0 is to impose floating point operations
-    if ((1.0 * (1 - dc_[0]) * dt_) / (1.0 * box_length_ * box_length_) >=
-        (1.0 / 6)) {
+    if (((1 - dc_[0]) * dt_) / (box_length_ * box_length_) >= (1.0 / 6)) {
       Log::Fatal(
           "DiffusionGrid",
           "The specified parameters of the diffusion grid with substance [",
           substance_name_,
           "] will result in unphysical behavior (diffusion coefficient = ",
           (1 - dc_[0]), ", resolution = ", resolution_,
           "). Please refer to the user guide for more information.");
+    } else if (diffusion_step_ <= 0.0 || diffusion_step_ > 10.0) {
+      Log::Fatal("DiffusionGrid",
+                 " The specified amount of diffusion steps for the grid with "
+                 "substance [",
+                 substance_name_,
+                 "] is not a real natural number between 1 to 10 please "
+                 "correct this and run the simulation again.");
```

Comment:
Should it be between 0 and 10, or between 1 and 10? The condition and error message do not agree. Please also explain in the documentation why this is required.